x-pack/plugins/security_solution/public/management/components/paginated_content/paginated_content.tsx

```diff
@@ -52,6 +53,8 @@ export interface PaginatedContentProps<T, C extends ComponentWithAnyProps> exten
   error?: ReactNode;
   /** Classname applied to the area that holds the content items */
   contentClassName?: string;
+  // Artifact specific decorations to display in the cards
+  CardDecorator: React.NamedExoticComponent<ArtifactEntryCardDecoratorProps> | undefined;
```

Comment:
FYI: you could have made this prop optional (`CardDecorator?: `), which would have avoided having to explicitly add the problem with `undefined` to the code as you did

Reply:
i deliberately choose this instead of having it optional, because this way you _must_ pass something - which helps avoid forgetting passing the prop with this kind of prop drilling. yesterday I forgot to add it in one place, and just noticed it was buggy while doing some exploratory tests today, so i wanted to ensure from types that the decorator is always forwarded in those components that just forward it

in the top level components (`ArtifactListPage` and `PolicyArtifactsLayout`) it is optional indeed